compiler/src/dotty/tools/dotc/semanticdb/TypeOps.scala

```diff
@@ -0,0 +1,344 @@
+            // `resType` of outer "inner")
+            //
+            // We try to find the "actual" binder of <y>: `inner`,
+            // and register them to the symbol table with `(<y>, inner) -> <y>`
+            // instead of `("y", outer) -> <y>`
+            if lam.paramNames.contains(sym.name) then
+              paramRefSymtab((lam, sym.name)) = sym
+            else
+              enterParamRef(lam.resType)
+
+          // for class constructor
+          // class C[T] { ... }
+          case cls: ClassInfo if sym.info.isInstanceOf[LambdaType] =>
+            val lam = sym.info.asInstanceOf[LambdaType]
+            cls.cls.typeParams.foreach { param =>
+              paramRefSymtab((lam, param.name)) = param
+            }
+
+          // type X[T] = ...
+          case tb: TypeBounds =>
+            enterParamRef(tb.lo)
+            enterParamRef(tb.hi)
+
+          case _ => ()
+        }
+
+      def enterRefined(tpe: Type): Unit =
+        tpe match {
+          case refined: RefinedType =>
+            val key = (refined, sym.name)
+            refinementSymtab(key) = sym
+
+          case rec: RecType =>
+            enterRefined(rec.parent)
+
+          case cls: ClassInfo
+            if (cls.cls.name == tpnme.REFINE_CLASS) =>
+            sym.owner.owner.info match {
+              // def foo(x: Parent { refinement }) = ...
+              case refined: RefinedType =>
+                enterRefined(refined)
+
+              // type x = Person { refinement }
+              case tb: TypeBounds =>
+                // tb = TypeBounds(
+                //   lo = RefinedType(...)
+                //   hi = RefinedType(...)
+                // )
+                enterRefined(tb.lo)
+                enterRefined(tb.hi)
+
+              // def s(x: Int): { refinement } = ...
+              case expr: ExprType =>
+                enterRefined(expr.resType)
+              case m: MethodType =>
+                enterRefined(m.resType)
+              case _ => ()
+            }
+          case _ => ()
+        }
+      enterParamRef(sym.owner.info)
+      enterRefined(sym.owner.info)
+
+      def loop(tpe: Type): s.Signature = tpe match {
+        case mp: MethodOrPoly =>
+          def flatten(
+            t: Type,
+            paramss: List[List[Symbol]],
+            tparams: List[Symbol]
+          ): (Type, List[List[Symbol]], List[Symbol]) = t match {
+            case mt: MethodType =>
+              val syms = mt.paramNames.flatMap { paramName =>
+                val key = (mt, paramName)
+                paramRefSymtab.get(key)
+              }
+              flatten(mt.resType, paramss :+ syms, tparams)
+            case pt: PolyType =>
+              val syms = pt.paramNames.flatMap { paramName =>
+                val key = (pt, paramName)
+                paramRefSymtab.get(key)
+              }
+              // there shouldn't multiple type params
```

Comment:
this assumption breaks with `Functor` example:
```scala
// tests/semanticdb/expect/Extension.scala
trait Functor[F[_]]:
  extension [T](t: F[T])
    def map[U](f: T => U): F[U]
```

```
ext/Functor#map(). => abstract method map [typeparam U ](param t: F[T])(param f: Function1[T, U]): F[U]
```

Reply:
Yeah... that's right.
The problem here is that we don't have a way to express multiple type params for a 
MethodSignature https://scalameta.org/docs/semanticdb/specification.html#signature

What we can do without extending SematnicDB would be grouping those type params into one type params like
`abstract method map [typeparam T, typeparam U ](param t: F[T])(param f: Function1[T, U]): F[U]`, but the things will be complicated when type `U` depends on type `T` as extension method's term parameter doesn.

Reply:
Done https://github.com/lampepfl/dotty/pull/12885/commits/05d31b85340c0c13985aecb5796b169e45d16c7f